### PR TITLE
Limit the maximum granularity of the time interval

### PR DESCRIPTION
**Pull request by karupayun**
Discussed once in the daily meeting. We agreed to have minimum 1 ns per
pixel to avoid visual issues but after seeing that we could have some
function calls of 50ns duration and some discussions, I decided to limit
100ns to the whole visible capture. If we try to zoom to a smaller
interval, we will center that interval in the minimum time range.

This PR is moving that new limit to the SetMinMax function to be
consistent. Before we had:

- 1 ns limit for zooming with mouse wheel
- no limit with ctrl + mouse right click

We have some issues when zooming-in a lot (Timers: http://b/168283058,
ThreadStates: http://b/200905941, and I even have a crash with ctrl + mouse
right click). They won't appear anymore after this PR.

PS: As a limitation, the overlay who is generated when holding mouse right
click is still rounded to an integer nanoseconds and this rounded is still
visible with 100ns total time range, as we can see in the screenshot:
http://screen/3x9qPTZWJsjnEQo. Anyway, it's not too drastic with 100ns and
I will consider that expected behaviour.

Test: several interactions in a loaded capture.

### CODE REVIEW

**Review by ronaldfw — APPROVED** [review]
LGTM, seems reasonable!

Nit: The PR description is duplicated for some reason. You should fix this in the commit message when merging.

**Comment by karupayun** [comment]
> LGTM, seems reasonable!
> 
> Nit: The PR description is duplicated for some reason. You should fix this in the commit message when merging.

Thanks Ronald! I was confused by why the commit message was so big but I didn't realize that was because duplicated message. Thanks for pointing it out! It's solved.

File: src/OrbitGl/TimeGraph.cpp [hunk 1]
```diff
@@ -176,8 +172,13 @@ void TimeGraph::VerticalZoom(float zoom_value, float mouse_normalized_y_position
 }
 
 void TimeGraph::SetMinMax(double min_time_us, double max_time_us) {
-  double desired_time_window = max_time_us - min_time_us;
-  min_time_us_ = std::max(min_time_us, 0.0);
+  const double kTimeGraphMinTimeWindowsUs = 0.1; /* 100 ns */
```
Comment (florian-kuebler) [inline src/OrbitGl/TimeGraph.cpp:175]: Nit:
```suggestion
  constexpr double kTimeGraphMinTimeWindowsUs = 0.1; /* 100 ns */
```

File: src/OrbitGl/TimeGraph.cpp [hunk 2]
```diff
@@ -176,8 +172,13 @@ void TimeGraph::VerticalZoom(float zoom_value, float mouse_normalized_y_position
...
+  double desired_time_window = std::max(max_time_us - min_time_us, kTimeGraphMinTimeWindowsUs);
```
Comment (florian-kuebler) [inline src/OrbitGl/TimeGraph.cpp:176]: Nit:
```suggestion
  const double desired_time_window = std::max(max_time_us - min_time_us, kTimeGraphMinTimeWindowsUs);
```

File: src/OrbitGl/TimeGraph.cpp [hunk 3]
```diff
@@ -176,8 +172,13 @@ void TimeGraph::VerticalZoom(float zoom_value, float mouse_normalized_y_position
...
+
+  // Centering the interval in screen.
+  double center_time_us = (max_time_us + min_time_us) / 2.;
```
Comment (florian-kuebler) [inline src/OrbitGl/TimeGraph.cpp:179]: Nit:
```suggestion
  const double center_time_us = (max_time_us + min_time_us) / 2.;
```

**Review by florian-kuebler — APPROVED** [review]
LGTM. Only really nit-picky comments. Thanks for the fix!

**Pull request merged**